src/script/interpreter.cpp

```diff
@@ -2012,6 +2041,8 @@ void PrecomputedTransactionData::Init(const T& txTo, std::vector<CTxOut>&& spent
         m_issuance_rangeproofs_single_hash = GetIssuanceRangeproofsSHA256(txTo);
         m_output_witnesses_single_hash = GetOutputWitnessesSHA256(txTo);
         m_spent_scripts_single_hash = GetSpentScriptsSHA256(m_spent_outputs);
+        m_spent_output_spk_single_hashes = GetSpentScriptPubKeysSHA256(m_spent_outputs);
```

Comment:
Shouldn't there be a std::move here or something?  Not that I know anything about move semantics.

Reply:
It should be fine. Any compiler since 2000 or so is smart enough to do "copy elision" when assigning the return value of a function like this.

Reply:
I was assuming that there would be some return value optimization that would apply the move semantics here. 
https://stackoverflow.com/questions/15704565/efficient-way-to-return-a-stdvector-in-c

Reply:
Updated the PR to use std::move

Reply:
Probably should turn it back.  It sounds like std:move for return values is somewhere between neutral and bad.

Reply:
Done